Valigator.AspNetCore/AspNetCore/VerifiableExtensions.cs

```diff
@@ -29,7 +29,7 @@ public static class VerifiableExtensions
 			=> value
 				.Match(
 					success => Option.Some(success.Match(
-						s => s is TValue ? success : throw new ValidateAttributeDoesNotSupportTypeException(s.GetType(), typeof(TValue)), 
+						s => s is TValue && !(s is TDataValue) ? success : throw new ValidateAttributeDoesNotSupportTypeException(s.GetType(), typeof(TValue)), 
```

Comment:
Won't this make it blow up on non-mapped non-nullable types? I don't think this needs to change.